### ISSUE TITLE
mpirun (master branch) error message for unknown and deprecated argument is wrong

### ISSUE THREAD

**Issue by wzamazon** [open-mpi/ompi#10097]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

master

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
git clone from source. then configured with `./configure --prefix=/fsx/ALinux2/PortaFiducia/libraries/openmpi/master/install --with-hwloc=external --disable-man-pages `

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.
```
94169f1725a26a156fd73fca1b61df14ccd581ab 3rd-party/openpmix (v1.1.3-3429-g94169f17)
 6722bd767e0fccbe07193f59f4f13e5610c7dd6a 3rd-party/prrte (psrvr-v2.0.0rc1-4274-g6722bd767e)
```

### Please describe the system on which you are running

* Operating system/version: Amazon Linux 2
* Computer hardware: Intel
* Network type: EFA

-----------------------------

## Details of the problem

The error messages of mpirun from master branch for deprecated/unknown argument is wrong.

For example:

If I pass a deprecated argument like `-machinefile`
```
/fsx/ALinux2/PortaFiducia/libraries/openmpi/master/install/bin/mpirun --map-by ppr:4:node -machinefile hostfile ./t
```

I got
```
--------------------------------------------------------------------------
An unrecognized option was included on the (null) command line:

  Option: ppr:4:node

Please use the "(null) --help" command to obtain a list of all
supported options.
--------------------------------------------------------------------------
prterun: command line error (Out of resource)
```
This is super confusing because "ppr:4:node" is valid input for `--map-by`. It is `-machinefile` that is causing trouble. So there seems to be an off-by-1 error in printing the arugment.

On the other hand, If I pass an unknown argument like `-i` like:
```
/fsx/ALinux2/PortaFiducia/libraries/openmpi/master/install/bin/mpirun -i
```
I got
```
--------------------------------------------------------------------------
An unrecognized option was included on the (null) command line:

  Option: -i

Please use the "(null) --help" command to obtain a list of all
supported options.
--------------------------------------------------------------------------
```
The error message is more helpful, though the `(null)` should be fixed.

**Comment by awlauria** [comment]
Working on a fix. Should post this morning

**Comment by awlauria** [comment]
Fix for unknown argument "(null)": 

https://github.com/openpmix/prrte/pull/1257
https://github.com/openpmix/openpmix/pull/2495

**Comment by awlauria** [comment]
Regarding the deprecation warning, it appears you are missing a `-` on `--machinefile`, which is causing the parser to get a bit wonky.

it seems to work when adding a double-dash.

FWIW it doesn't seem to be in prrte's deprecated list. I don't think it is deprecated.

**Comment by wzamazon** [comment]
> it appears you are missing a - on -machinefile, which is causing the parser to get a bit wonky.

Ah, I see. Thank you!

However, single dash is accepted by open mpi 4, so I think it should be handled by v5 (or at least print a proper warning).

> We could add single-dash machinefile to the list of those converted from single-dash to double dash (or convert all of them..). I feel like prrte used to do that, but it got dropped at some point?

Yes, I remember prrte used to do that.

**Comment by awlauria** [comment]
No worries, I can try posting that patch, but may not until next week.

The fixes for "(null)" are fixed in pmix/prte master, just need to update the submodule.

**Comment by wzamazon** [comment]
I can confirm the "(null)" has been fixed by latest pmix/prte. Thank you!

**Comment by awlauria** [comment]
@wzamazon I opened https://github.com/openpmix/prrte/pull/1269 to try and fix the single dash issue if you want to test it.

**Comment by wzamazon** [comment]
I can confirm https://github.com/openpmix/prrte/pull/1269 fixed the issue of single dash

**Comment by awlauria** [comment]
This should be fixed in master with #10136 fixed in master. Needs back porting to v5.0.0x

**Comment by awlauria** [comment]
This should be fixed in the v5.0.x branch via the following submodule pointer update: https://github.com/open-mpi/ompi/pull/10161

for master: https://github.com/open-mpi/ompi/pull/10165 (This PR fixes a variety of issues, but updating the submodule here will fix this issue on main).
